snapcraft/internal/pluginhandler/_dependencies.py

```diff
@@ -0,0 +1,92 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2019 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+import os
+from typing import Sequence, Set
+
+from snapcraft.internal import repo
+
+
+_MSG_EXTEND_STAGE_PACKAGES = (
+    "The {part_name!r} part is missing libraries that are not "
+    "included in the snap or base. They can be satisfied by adding the following "
+    "entries to the existing stage-packages for this "
+    "part:\n{stage_packages}"
+)
+_MSG_ADD_STAGE_PACKAGES = (
+    "The {part_name!r} part is missing libraries that are not "
+    "included in the snap or base. They can be satisfied by adding the following "
+    "entry for this part\nstage-packages:\n{stage_packages}"
+)
+_MSG_UNHANDLED_DEPENDENCIES = (
+    "This part is missing libraries that cannot be satisfied with "
+    "any available stage-packages known to snapcraft:\n{unhandled_list}\n"
+    "These dependencies can be satisfied via more parts (ensure filesets are correct), "
```

Comment:
NIT: (ensure filesets are correct) ... 

perhaps:
"...via additional parts or content sharing.  Consider validating configured filesets if this dependency built."